### PR TITLE
Upgrade jsonwebtoken to 7.0 and re-enable

### LINKED ISSUE #390: Re-enable JWT middleware after async/await
The JWT middleware was disabled in #370 and is still blocked on an updated version of jsonwebtoken due to a conflicting dependency on ring. It doesn't seem to be maintained: https://github.com/Keats/jsonwebtoken#help-wanted-for-v7

### PR DESCRIPTION
**Pull request by colinbankier**
Fixes #390

### CODE REVIEW

**Comment by pksunkara** [comment]
#401 fixes the fmt issues and there's a small issue with docs test. Otherwise, looks great.

File: middleware/jwt/src/middleware.rs [hunk 1]
```diff
@@ -145,10 +147,14 @@ where
...
             trace!("[{}] bad request jwt middleware", request_id(&state));
             let res = create_empty_response(&state, StatusCode::BAD_REQUEST);
-            return Box::new(future::ok((state, res)));
+            return future::ok((state, res)).boxed();
         }
 
-        match decode::<T>(&token.unwrap(), self.secret.as_ref(), &self.validation) {
+        match decode::<T>(
+            &token.unwrap(),
+            &DecodingKey::from_secret(self.secret.as_ref()),
```
Comment (whitfin) [inline middleware/jwt/src/middleware.rs]: I think this line would be better outside and stored as a local

Comment (colinbankier) [reply middleware/jwt/src/middleware.rs]: Yep, done.

**Comment by colinbankier** [comment]
@nyarly or @whitfin - please give this a merge if you see no issues.

**Pull request merged**